projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+*      of one or more patent holders.  This license does not release you
+*      from the requirement that you obtain separate licenses from these
+*      patent holders to use this software.
+*    - Use of the software either in source or binary form, must be run
+*      on or directly connected to an Analog Devices Inc. component.
+*
+*THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+*PARTICULAR PURPOSE ARE DISCLAIMED.
+*
+*IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+*RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+*BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+*THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************
+**/
+
+#include <sys/platform.h>
+#include "adi_initialize.h"
+#include "ADuCM3029_demo_cn0428_cn0429.h"
+#include "adi_m355_gas_sensor.h"
+#include <stdio.h>
+#include <string.h>
+extern "C" {
+#include "RingBuffer.h"
+#include "cli.h"
+}
+
+using namespace adi_sensor_swpack;
+
+/* ================================================== */
+/* 	SYSTEM VARIABLES - handles, memory, error flags	  */
+/* ================================================== */
+ADI_UART_RESULT		  eUartResult;
+ADI_GPIO_RESULT 	  eGpioResult;
+ADI_PWR_RESULT        ePwrResult;
+SENSOR_RESULT   	  eSensorResult;
+
+/* Memory for the GPIO driver */
+uint8_t gpioMemory[ADI_GPIO_MEMORY_SIZE] = {0};
+
+/* Handle for the UART device. */
+ADI_UART_HANDLE hDevice;
+
+/* Memory for the UART driver. */
+static uint8_t UartDeviceMem[ADI_UART_MEMORY_SIZE];
+
+/* ================================================== */
+/* 						VARIABLES					  */
+/* ================================================== */
+const ADI_GPIO_PORT CS_Port [4] = {ADI_GPIO_PORT0, ADI_GPIO_PORT1, ADI_GPIO_PORT2, ADI_GPIO_PORT0};		// Store Chip Select GPIO Port address
+const ADI_GPIO_DATA CS_Pin [4] = {ADI_GPIO_PIN_8, ADI_GPIO_PIN_11, ADI_GPIO_PIN_1, ADI_GPIO_PIN_9};		// Store Chip Select GPIO Pin address
+const ADI_GPIO_PORT INT_Port [4] = {ADI_GPIO_PORT0, ADI_GPIO_PORT0, ADI_GPIO_PORT1, ADI_GPIO_PORT1};	// Store Interrupt GPIO Port address
+const ADI_GPIO_DATA INT_Pin [4] = {ADI_GPIO_PIN_15, ADI_GPIO_PIN_13, ADI_GPIO_PIN_12, ADI_GPIO_PIN_14};	// Store Interrupt GPIO Pin address
+
+/* Ringbuffer structure */
+struct RingBuf RX, TX;
+
+const uint8_t M355_Demo_Msg[64]		= "-------- M355 Gas Sensing Demo --------";
+const uint8_t M355_Demo_Msg_Wq[64]	=
+	"-------- M355 Water Quality Demo --------";
```

Comment:
Please move the definition on one line.

Reply:
This was also done by AStyle. I'll try to fix it again and check if there are others like this.

Reply:
funny thing actually, i don't think these variables are used anywhere ?
typically const-strings are just printed directly via printf somewhere; like `printf( "-------- M355 Gas Sensing Demo --------")` ;
there's not a whole lot of reason to keep them in a const entity ; you'd typically either have a `print_demo_msg()` function that you call, that does the `printf()`; that type of construct is a bit more flexible, as you can also parametrize stuff [if needed]

Reply:
but as another note here: yeah, this is one place where astyle doesn't shine [when enforcing 80 columns] ;
every once in a while, i keep wondering about whether we still need the 80 column restriction-rule [in code], or whether we should bump that higher [to 120 or something], but then i keep coming back to the kernel coding guidelines ;
[ FWIW, a lot of our coding guidelines are derived loosely from the kernel coding guidelines ; we can [and sometimes do] slightly deviate from them, unless pushing code to the mainline kernel ]

regarding indentation & long-lines:
https://www.kernel.org/doc/html/v4.10/process/coding-style.html#indentation